include/ddwaf.h

```diff
@@ -143,6 +143,23 @@ struct _ddwaf_version
     uint16_t patch;
 };
 
+/**
+ * @ddwaf_ruleset_info
+ *
+ * Structure containing diagnostics on the provided ruleset.
+ * */
+struct _ddwaf_ruleset_info
+{
+    /** Number of rules successfully loaded **/
+    uint16_t loaded;
+    /** Number of rules which failed to parse **/
+    uint16_t failed;
+    /** Map of rule parsing errors {error: [rules]} **/
```

Comment:
I think describing it as _a map from an error string to an array of all the rule ids for which that error was raised_ would be clearer.